File: ts-sdk/src/sdk/models/operations/listusers.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class ListUsersQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=limit" })
  limit?: number;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=offset" })
  offset?: number;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=false;name=org_ids" })
  orgIds?: string[];

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=query" })
  query?: string;
}


export class ListUsers200ApplicationJson extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=users", elemType: shared.User })
  users?: shared.User[];
}


export class ListUsersRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  queryParams: ListUsersQueryParams;
}


export class ListUsersResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  listUsers200ApplicationJSONObject?: ListUsers200ApplicationJson;
}

File: go-sdk/pkg/models/operations/listusers.go
package operations

import (
	"openapi/pkg/models/shared"
)

type ListUsersQueryParams struct {
	Limit  *float64 `queryParam:"style=form,explode=true,name=limit"`
	Offset *float64 `queryParam:"style=form,explode=true,name=offset"`
	OrgIds []string `queryParam:"style=form,explode=false,name=org_ids"`
	Query  *string  `queryParam:"style=form,explode=true,name=query"`
}

type ListUsers200ApplicationJSON struct {
	Users []shared.User `json:"users,omitempty"`
}

type ListUsersRequest struct {
	QueryParams ListUsersQueryParams
}

type ListUsersResponse struct {
	ContentType                       string
	StatusCode                        int64
	ListUsers200ApplicationJSONObject *ListUsers200ApplicationJSON
}
